cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+"""
+CmdStan arguments
+"""
+import os
+import tempfile
+from typing import List, Union
+
+import numpy as np
+
+from cmdstanpy.utils import check_csv, read_metric
+
+
+class SamplerArgs(object):
+    """Arguments for the NUTS adaptive sampler."""
+
+    def __init__(
+        self,
+        warmup_iters: int = None,
+        sampling_iters: int = None,
+        save_warmup: bool = False,
+        thin: int = None,
+        max_treedepth: int = None,
+        metric: Union[str, List[str]] = None,
+        step_size: Union[float, List[float]] = None,
+        adapt_engaged: bool = None,
+        adapt_delta: float = None,
+    ) -> None:
+        """Initialize object."""
+        self.warmup_iters = warmup_iters
+        self.sampling_iters = sampling_iters
+        self.save_warmup = save_warmup
+        self.thin = thin
+        self.max_treedepth = max_treedepth
+        self.metric = metric
+        self.metric_file = None
+        self.step_size = step_size
+        self.adapt_engaged = adapt_engaged
+        self.adapt_delta = adapt_delta
+
+    def validate(self, chains: int) -> None:
+        """
+        Check arguments correctness and consistency.
+
+        * adaptation and warmup args are consistent
+        * if file(s) for metric are supplied, check contents.
+        * length of per-chain lists equals specified # of chains
+        """
+        if type(chains) is not int or chains < 1:
```

Comment:
the correct way to test this is

    isinstance(chains, int)

but for example numpy np.int64 != int, so the safest way to test this would be 

    from numbers import Integral
    ...
    ... isinstance(chains, Integral)

Reply:
good catch - thanks - fixed.